mail_connector_queue/job.py

```diff
@@ -18,6 +18,8 @@ def send_mail(session, mail_id):
 
 @on_record_create(model_names='mail.mail')
 def mail_creation(session, model_name, record_id, vals):
+    if vals.get('state') != 'outgoing':
+        return
```

Comment:
the default is outgoing (I think, to be checked), so `vals.get('state', 'outgoing')`